renderdoc/driver/metal/metal_device.cpp

```diff
@@ -38,3 +51,121 @@ MTL::Device *WrappedMTLDevice::MTLCreateSystemDefaultDevice(MTL::Device *realMTL
 
   return UnwrapObjC<MTL::Device *>(wrappedMTLDevice);
 }
+
+MTL::Library *WrappedMTLDevice::newDefaultLibrary()
+{
+  MTL::Library *realMTLLibrary;
+
+  SERIALISE_TIME_CALL(realMTLLibrary = Unwrap<MTL::Device *>(this)->newDefaultLibrary());
```

Comment:
You should improve the `Unwrap` template so that it type detects the correct unwrap, rather than needing an explicit template parameter like this.